common/testing/taskpoller/taskpoller.go

```diff
@@ -0,0 +1,315 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package taskpoller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	failurepb "go.temporal.io/api/failure/v1"
+	historypb "go.temporal.io/api/history/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/debug"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/rpc"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/service/history/consts"
+)
+
+type (
+	TaskPoller struct {
+		t          require.TestingT
+		_logger    log.Logger
+		_client    workflowservice.WorkflowServiceClient
+		_namespace string
+	}
+	Options struct {
+		ctx                 context.Context
+		tv                  *testvars.TestVars
+		maxRetries          int
+		pollStickyTaskQueue bool
+	}
+	// OptionFunc is a function to change an Options instance
+	OptionFunc func(*Options)
+)
+
+var (
+	// DrainTaskFunc returns an empty RespondWorkflowTaskCompletedRequest response
+	DrainTaskFunc = func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondWorkflowTaskCompletedRequest, error) {
```

Comment:
```suggestion
	DrainTaskFn = func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondWorkflowTaskCompletedRequest, error) {
```
I think the convention is `Func` for types, `Fn` for vars.

Reply:
I think `Func` is more idiomatic (ChatGPT agrees); double-checking ...

Reply:
In our codebase I'm seeing 36 hits for `Fn = ` and 66 for `Func = `